subxt/src/client/lightclient/builder.rs

```diff
@@ -0,0 +1,201 @@
+// Copyright 2019-2023 Parity Technologies (UK) Ltd.
+// This file is dual-licensed as Apache-2.0 or GPL-3.0.
+// see LICENSE for license details.
+
+use super::{rpc::LightClientRpc, LightClient, LightClientError};
+use crate::{config::Config, error::Error, OnlineClient};
+
+#[cfg(feature = "jsonrpsee")]
+use jsonrpsee::{
+    async_client::ClientBuilder,
+    client_transport::ws::{Uri, WsTransportClientBuilder},
+    core::client::ClientT,
+    rpc_params,
+};
+use smoldot_light::ChainId;
+use std::num::NonZeroU32;
+use std::sync::Arc;
+
+/// Builder for [`LightClient`].
+#[derive(Clone, Debug)]
+pub struct LightClientBuilder {
+    max_pending_requests: NonZeroU32,
+    max_subscriptions: u32,
+    bootnodes: Option<Vec<String>>,
+    potential_relay_chains: Option<Vec<ChainId>>,
+}
+
+impl Default for LightClientBuilder {
+    fn default() -> Self {
+        Self {
+            max_pending_requests: NonZeroU32::new(128)
+                .expect("Valid number is greater than zero; qed"),
+            max_subscriptions: 1024,
+            bootnodes: None,
+            potential_relay_chains: None,
+        }
+    }
+}
+
+impl LightClientBuilder {
+    /// Create a new [`LightClientBuilder`].
+    pub fn new() -> LightClientBuilder {
+        LightClientBuilder::default()
+    }
+
+    /// Overwrite the bootnodes of the chain specification.
+    ///
+    /// Can be used to provide trusted entities to the chian spec, or for
+    /// testing environments.
+    pub fn bootnodes<'a>(mut self, bootnodes: impl IntoIterator<Item = &'a str>) -> Self {
```

Comment:
could be a good idea to make this fallible and decode as json values here directly instead of detecting that when the chainspec has been downloaded....

Reply:
should probably be infallible anyway so shouldn't matter I guess..

Reply:
Oh yeap, this simplifies the code quite a bit!
Have stored them as `Vec<serde_json::Value>` in the builder directly, thanks!